shared/libraries/websocket_streaming/tests/mock_streaming_server.cpp

```diff
@@ -0,0 +1,362 @@
+    {
+        return onControlCommand(streamId, command, signalIds, errorMessage);
+    };
+    this->controlServer =
+        std::make_unique<daq::streaming_protocol::ControlServer>(ioContext,
+                                                                 controlPort,
+                                                                 controlCommandCb,
+                                                                 logCallback);
+    this->controlServer->start();
+
+    this->serverThread = std::thread([this]() { this->ioContext.run(); });
+}
+
+void MockStreamingServer::stop()
+{
+    ioContext.stop();
+
+    if (!serverThread.joinable())
+        return;
+
+    this->server->stop();
+    serverThread.join();
+    this->server.reset();
+}
+
+void MockStreamingServer::onAccept(const OnAcceptCallback& callback)
+{
+    onAcceptCallback = callback;
+}
+
+void MockStreamingServer::onAcceptInternal(const daq::stream::StreamPtr& stream)
+{
+    auto writer = std::make_shared<StreamWriter>(stream);
+    writeProtocolInfo(writer);
+    writeInit(writer);
+
+    auto signals = List<ISignal>();
+    if (onAcceptCallback)
+        signals = onAcceptCallback();
+
+    auto outputSignals = std::unordered_map<std::string, std::pair<SizeT, SignalPtr>>();
+    SizeT signalNumber = 0;
+    for (const auto& signal : signals)
+    {
+        bool isDomainSignal = !signal.getDomainSignal().assigned();
+        auto descriptor = signal.getDescriptor();
+
+        if (!descriptor.assigned() ||
+            !isDomainSignal && descriptor.getSampleType() != daq::SampleType::Float64 ||
+            !isDomainSignal && descriptor.getRule().getType() != daq::DataRuleType::Explicit ||
+            isDomainSignal && descriptor.getSampleType() != daq::SampleType::UInt64 ||
+            isDomainSignal && descriptor.getRule().getType() != daq::DataRuleType::Linear)
+        {
+            LOG_E("Unsupported type of signal {}; data signal should be explicit float64, "
+                  "domain signal linear unsigned64", signal.getGlobalId());
+            throw GeneralErrorException("Unsupported type of signal");
+        }
+        outputSignals.insert({signal.getGlobalId().toStdString(), {++signalNumber, signal}});
+    }
+
+    LOG_I("New client connected. Stream Id: {}", writer->id());
+    clients.insert({writer, outputSignals});
+
+    writeSignalsAvailable(writer, signals);
+}
+
+int MockStreamingServer::onControlCommand(const std::string& streamId,
+                                          const std::string& command,
+                                          const daq::streaming_protocol::SignalIds& signalIds,
+                                          std::string& errorMessage)
+{
+    if (signalIds.empty())
+    {
+        LOG_W("Signal list is empty, reject command", streamId);
+        errorMessage = "Signal list is empty";
+        return -1;
+    }
+
+    auto clientIter = std::find_if(std::begin(clients),
```

Comment:
Since this is a Mock, no action needs to be taken here. I am just wondering if would it be possible to have a constant time lookup by the streamId?

Reply:
The real server had the identical implementation, so I've made adjustments for optimization across both. The `streamId` is now utilized as the key, ensuring a constant lookup time.